zad2/warriors.rb

```diff
@@ -1,12 +1,122 @@
-class Warrior
-  def initialize(name:, level:)
+class Character
+  attr_accessor :name, :level
+
+  #Constructor
+  def initialize(name:,level:)
+    @name = name
+    @level = is_valid?(level)
+  end
+
+  #Card displays all the attributes of our character
+  def card
+  puts "#{name} is #{level} level"
   end
 
+  #Strength method calculates the damage dealt value
   def strength
+    [*1..12].sample + @level
+  end
+
+  protected
+
+  #Finding out if the level is valid (it is supposed to be between 1 and 99)
+  def is_valid?(valid_level)
+    valid_level.between?(1,99) ? valid_level : @level = 1
   end
 end
 
+class Warrior < Character
+  #Warrior is the only class able to level up, the method below gives him this possibility
+  def lvl_up(winner,loser)
+    if winner.level >= loser.level
+      winner.level += 1
+    else
+      winner.level = loser.level - winner.level + 1
+    end
+  end
+end
+
+class Monster < Character
+end
+
 class BattleArena
+  attr_accessor :first_character, :second_character
+
   def initialize(first_character, second_character)
+    self.first_character = first_character
+    self.second_character = second_character
+  end
+
+  #Calling out battle force the characters to fight
+  def battle!
+    first_dmg = first_character.strength
+    second_dmg = second_character.strength
+
+    puts "\n#{first_character.name} attacked #{second_character.name} with #{first_dmg} damage"
+    puts "#{second_character.name} attacked #{first_character.name} with #{second_dmg} damage"
+    
+    #Let's check which character is victorious
+    if first_dmg > second_dmg
+      winner = first_character
+
+      #Checking if the winner of our fight is a warrior (then lvl him up), it excludes monsters to get exp 
+      winner.is_a?(Warrior) ? winner.lvl_up(first_character,second_character) : winner.level = winner.level
```

Comment:
`condition ? do_a : do_b` to zwykły alias do `if/else`, a `else` w tym przypadku nie potrzebny. Proponuję zatem: 
```
winner.lvl_up(first_character,second_character) if winner.is_a?(Warrior)
```